docs/lib/graphqlapi/fragments/ios/query-data.md

```diff
@@ -121,3 +119,87 @@ func listTodos() -> AnyCancellable {
 </amplify-block>
 
 </amplify-block-switcher>
+
+### List subsequent pages of items
+
+For large data sets, you'll need to paginate through the results. After receiving the first page of results, you can check if there is a subsequent page and obtain the next page.
+
+```swift
+var todos: [Todo] = []
+var currentPage: List<Todo>?
+
+func listTodos() {
+    let todo = Todo.keys
+    let predicate = todo.name == "my first todo" && todo.description == "todo description"
+    Amplify.API.query(request: .paginatedList(Todo.self, where: predicate, limit: 1000)) { event in
+        switch event {
+        case .success(let result):
+            switch result {
+            case .success(let todos):
+                print("Successfully retrieved list of todos: \(todos)")
+                self.currentPage = todos
+                self.todos.append(contentsOf: todos)
+            case .failure(let error):
+                print("Got failed result with \(error.errorDescription)")
+            }
+        case .failure(let error):
+            print("Got failed event with error \(error)")
+        }
+    }
+}
+
+func loadMore() {
+    if let current = currentPage, current.hasNextPage() {
+        current.getNextPage { result in
+            switch result {
+            case .success(let todos):
+                self.todos.append(contentsOf: todos)
+                self.currentPage = todos
+            case .failure(let coreError):
+                print("Failed to get next page \(coreError)")
+            }
+        }
+    }
+}
+```
+
+## List all pages
+
+If you want to get all pages, first get the initial page and then use it to recursively check if there is a subsequent page before retrieving the next page, aggregating the data from each page into the `todos` array.
+
+```swift
+var todos: [Todo] = []
+var currentPage: List<Todo>?
+
+func getAllPages() {
```

Comment:
Seems like it would be just as easy to reuse the code above, but change the success handlers in both the "initial" and "next" methods to invoke the "loadNext" method? Fewer concepts, and arguably better separation of concerns. I'm not super happy with the amount of work we've got going on in this method.

Reply:
ah nice, i've refactored to use the above code, and updated the `listNextPage` to be called `listNextPageRecursively`